src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -423,6 +429,9 @@ public bool IsLayoutFixedInCurrentVersionBubble(TypeDesc type)
             return true;
         }
 
+        public bool IsLayoutFixedInCurrentVersionBubble(TypeDesc type) =>
+            _computedFixedLayoutTypes.GetOrAdd(type, (_) => IsLayoutFixedInCurrentVersionBubbleInternal(type));
```

Comment:
This avoids allocating the capture - we still allocate the delegate (and don't really have to), but it's a nice 50% reduction:

```suggestion
            _computedFixedLayoutTypes.GetOrAdd(type, (t) => IsLayoutFixedInCurrentVersionBubbleInternal(t));
```

Reply:
Also, `GetOrAdd` [is non-atomic](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=netcore-3.1) as the write operation is not thread-safe. The delegate can execute multiple times for the same key, which can be prevented by something like this:

```c#
_computedFixedLayoutTypes.GetOrAdd(type, t =>
{
    Interlocked.Increment(ref s_someStaticInteger);
    return _computedFixedLayoutTypes.TryGetValue(t, out bool isLayoutFixed) ?
        isLayoutFixed : IsLayoutFixedInCurrentVersionBubbleInternal(t));
});
```

Reply:
We generally don't worry about those kinds of races - the worst that can happen is that we compute the same thing twice. 95% of time the thing we're computing is actually cheap to compute anyway so chances of that mattering are pretty low.

I'm not sure the double-check would actually guarantee we don't compute the same thing twice - the delegate is called outside locks, so we could still end up computing the same things twice.

Reply:
> the worst that can happen is that we compute the same thing twice.

Typically, it is proportional to degree of parallelism.
>The delegate is called outside locks, so we could still end up computing the same things twice.

The lock before second check will prevent it from happening.

Reply:
Thanks Michal; I removed the capture. I agree here we're not that worried about a race in this situation. The value returned by `IsLayoutFixedInCurrentVersionBubbleInternal` is going to be identical throughout a given compilation for a given `TypeDesc`, too